ReactWindows/ReactNative.Shared/Views/Control/ReactControlShadowNode.cs

```diff
@@ -0,0 +1,23 @@
+using Facebook.Yoga;
+using Newtonsoft.Json.Linq;
+using ReactNative.Reflection;
+using ReactNative.UIManager.Annotations;
+using System;
+using static System.FormattableString;
+
+namespace ReactNative.UIManager
+{
+    /// <summary>
+    /// Shadow node subclass for WindowsControl.
+    /// </summary>
+    public class ReactControlShadowNode : LayoutShadowNode
+    {
+        /// <summary>
+        /// Instantiates a <see cref="ReactControlShadowNode"/>.
+        /// </summary>
+        public ReactControlShadowNode() :
+            base(false, true)
```

Comment:
>true [](start = 24, length = 4)

I see now the need for the "delegated" layout - not sure this is the best approach to achieve what you're going for here (not saying it isn't - just saying I haven't though about how to best handle pure wrapper view layout before)

Reply:
It all boils down to a component trying to "split" the styles it receives in its props to the two elements it consists of: the FocusableViewWindows and whatever child.
I tried various things like keeping the "apply to component" ones (like "align self") to FocusableViewWindows and pass the others to the child, and lots of other combinations.
Nothing works fully, partially because some of the styles passed to the child depend on the context sent by parent, and that's something I don't know (what should be the flex-direction set on FocusableViewWindows? Hard to guess)
That's how I got to this solution, though I missed the non style props that do have visual position effects (the "transform"). I could go a more elegant way, but I don't know any at the moment.
For the transform, I could revive some code I transiently had, one that flattens the styles, extracts a set (transform in this case) to keep it to FVW, etc. I'll see, still thinking...

Reply:
And yes, using a style splitter in JS land fixes the problem. But that's so unfriendly to whoever uses this construct, yuck. Still thinking of some magic bullet in the native land.

Reply:
It all boils down to a component trying to "split" the styles it receives in its props to the two elements it consists of: the FocusableViewWindows and whatever child.
I tried various things like keeping the "apply to component" ones (like "align self") to FocusableViewWindows and pass the others to the child, and lots of other combinations.
Nothing works fully, partially because some of the styles passed to the child depend on the context set by parent, and that's something I don't know (what should be the flex-direction set on FocusableViewWindows? Hard to guess)
That's how I got to this solution, though I missed the non layout props that do have visual position effects (the "transform"). I could go a more elegant way, but I don't know any at the moment.
For the transform, I could revive some code I transiently had, one that flattens the styles, extracts a set (transform in this case) to keep it for FVW, etc. I'll see, still thinking...



---
In reply to: [155706780](https://github.com/Microsoft/react-native-windows/pull/1546#discussion_r155706780) [](ancestors = 155706780)